libraries/botbuilder-dialogs-adaptive/src/actions/cancelAllDialogs.ts

```diff
@@ -6,40 +6,57 @@
  * Licensed under the MIT License.
  */
 import { DialogTurnResult, DialogContext, DialogConfiguration, Dialog } from 'botbuilder-dialogs';
+import { ExpressionEngine } from 'botframework-expressions';
 
 export interface CancelAllDialogsConfiguration extends DialogConfiguration {
     eventName?: string;
     eventValue?: string;
-    eventValueProperty?: string;
 }
 
 export class CancelAllDialogs<O extends object = {}> extends Dialog<O> {
 
-    constructor();
-    constructor(eventName: string, eventValue?: string);
-    constructor(eventName?: string, eventValue?: string) {
+    public static declarativeType = 'Microsoft.CancelAllDialogs';
+
+    public constructor();
+    public constructor(eventName: string, eventValue?: string);
+    public constructor(eventName?: string, eventValue?: string) {
         super();
         this.eventName = eventName;
-        this.eventValue = eventValue
-    }
-
-    protected onComputeId(): string {
-        return `CancelAllDialogs[${this.eventName || ''}]`;
+        this.eventValue = eventValue;
     }
 
+    /**
+     * Event name.
+     */
     public eventName: string;
 
+    /**
+     * Value expression for event value.
+     */
     public eventValue: string;
 
     public configure(config: CancelAllDialogsConfiguration): this {
         return super.configure(config);
     }
 
     public async beginDialog(dc: DialogContext, options: O): Promise<DialogTurnResult> {
-        const opt = Object.assign({
-            eventName: this.eventName,
-            eventValue: this.eventValue
-        }, options);
-        return await dc.cancelAllDialogs(true, opt.eventName, opt.eventValue);
+        let eventValue: any;
+        if (this.eventValue) {
+            const eventValueExpression = new ExpressionEngine().parse(this.eventValue);
```

Comment:
> new ExpressionEngine().parse(this.eventValue); [](start = 40, length = 47)

Again... why are we re-parsing the expression every time we evaluate it?

Reply:
> > new ExpressionEngine().parse(this.eventValue); [](start = 40, length = 47)
> 
> Again... why are we re-parsing the expression every time we evaluate it?

I'll cache the parsed expression in a private property.